quartz/util/imageHelper.ts

```diff
@@ -0,0 +1,126 @@
+import { FontWeight, SatoriOptions } from "satori/wasm"
+import { GlobalConfiguration } from "../cfg"
+import { JSXInternal } from "preact/src/jsx"
+import { QuartzPluginData } from "../plugins/vfile"
+
+/**
+ * Get an array of `FontOptions` (for satori) given google font names
+ * @param headerFontName name of google font used for header
+ * @param bodyFontName name of google font used for body
+ * @returns FontOptions for header and body
+ */
+export async function getSatoriFont(headerFontName: string, bodyFontName: string) {
+  const headerWeight = 700 as FontWeight
+  const bodyWeight = 400 as FontWeight
+
+  // Fetch fonts
+  const headerFont = await fetchTtf(headerFontName, headerWeight)
```

Comment:
Hmm, given that this only works for google fonts, we should probably say somewhere that if you use local fonts (saved under `static`) then users might have to edit this file

https://github.com/jackyzha0/quartz/blob/0ad9111388dc6933cec6e22499d15de0dc156c05/quartz.config.ts#L23

an example would be

```ts
import { joinSegments } from "./path"

const headerFontPath = joinSegments("static", "GT-Sectra-Display-Regular.woff")
const bodyFontPath = joinSegments("static", "GT-Sectra-Book.woff")

...
const url = new URL(`https://${cfg.baseUrl ?? "example.com"}`)

const [headerFont, bodyFont] = await Promise.all(
  [headerFontPath, bodyFontPath].map((font) =>
    fetch(`${url.toString()}/${font}`).then((res) => res.arrayBuffer()),
  ),
)
```

Reply:
Hmm making users edit source code for configuration seems to go against the spirit of quartz a little bit, I guess ideally this would be another config option like `localHeaderFont` and `localBodyFont` where users could just pass in the font file names and the rest would get resolved on its own (either using the `fetchTtf` function if `fontOrigin: "googleFonts"` or using the two new properties if set to local).

This shouldn't be too hard to implement, have you tested if the code you provided works like that? If it does, I'll try to integrate this before we merge. I also noticed a leftover review from jacky where fonts get loaded even if the config is disabled due to a guard thats too far down, I'll also try to fix that.

Reply:
Oh thats unfortunate. I think I wrote the PR before `fontOrigin` was even a thing so it defaulted to only using google fonts, but doing it semi-automatic might still be more convenient? We could provide an example of how to add a font face to `customs.scss`, but there's less potential for things to go wrong there as you can just add it on the bottom or wherever, vs doing actual changes to source code?